nms/app/packages/magmalte/app/components/feg/FEGContext.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @flow strict-local
+ * @format
+ */
+
+import * as React from 'react';
+import FEGNetworkContext from '../context/FEGNetworkContext';
+import LoadingFiller from '@fbcnms/ui/components/LoadingFiller';
+import MagmaV1API from '@fbcnms/magma-api/client/WebClient';
+
+import type {feg_network, network_id, network_type} from '@fbcnms/magma-api';
+
+import {UpdateNetworkState as UpdateFegNetworkState} from '../../state/feg/NetworkState';
+
+import {useEffect, useState} from 'react';
+import {useEnqueueSnackbar} from '@fbcnms/ui/hooks/useSnackbar';
+
+type Props = {
+  networkId: network_id,
+  networkType: network_type,
+  children: React.Node,
+};
+import {FEG} from '@fbcnms/types/network';
+
+/**
+ * Fetches and returns information about the federation network inside
+ * a context provider.
+ * @param {object} props: contains the network id and its type
+ */
+export function FEGNetworkContextProvider(props: Props) {
+  const {networkId} = props;
+  const [fegNetwork, setFegNetwork] = useState<$Shape<feg_network>>({});
+  const [isLoading, setIsLoading] = useState(true);
+  const enqueueSnackbar = useEnqueueSnackbar();
+
+  useEffect(() => {
+    const fetchState = async () => {
```

Comment:
I tried to change it so that we use useMagmaAPI, but it turns out that you can't use that if you are setting the response you get from that function into a react state variable because after you get the response and set that response to a react state variable it re-renders the page and this process goes on and on making the page to rerender infinite times. If I was using this in a simple page(like in Subscribers.js file), and I dont need to save the response into a state variable this was fine but since I need to set the response to a state variable because this is a context provider, unfortunately, I can't use that function.

Reply:
> I tried to change it so that we use useMagmaAPI, but it turns out that you can't use that if you are setting the response you get from that function into a react state variable because after you get the response and set that response to a react state variable it re-renders the page and this process goes on and on making the page to rerender infinite times. If I was using this in a simple page(like in Subscribers.js file), and I dont need to save the response into a state variable this was fine but since I need to set the response to a state variable because this is a context provider, unfortunately, I can't use that function.

You don't need to set the response to the state variable. Instead you use a callback pattern as follows. Look at the example here -> https://github.com/magma/magma/blame/d691319dd40e0a7d2822a993f819beca7490e65d/nms/app/packages/magmalte/app/components/admin/FEGNetworkDialog.js#L49

Reply:
Is it completely unavoidable without simple changes? I'm referencing [this](https://daveceddia.com/useeffect-hook-examples/#prevent-useeffect-from-running-every-render)

Reply:
Oh, I have changed it now using the useCallback function.